openrouteservice/src/main/java/org/os_concavehull/ConcaveHull.java

```diff
@@ -0,0 +1,1037 @@
+			node = node.getNext();
+		} while (node != hullCL.getNode());
+	}
+
+	private void generateHullIndices(DLCirList<Coordinate> hullCL, TriangleMetric m, Map<Coordinate, DLNode<Coordinate>> coordNodeMap, Set<HullEdgeCir> edgeIdx, Map<DLNode<Coordinate>, HullEdgeCir> nodeEdgeMap) {
+		DLNode<Coordinate> node = hullCL.getNode();
+		do {
+			Coordinate coord = node.getObj();
+			coordNodeMap.put(coord, node);
+			HullEdgeCir e = new HullEdgeCir(node, m);
+			edgeIdx.add(e);
+			nodeEdgeMap.put(node, e);
+			node = node.getNext();
+		} while (node != hullCL.getNode());
+
+	}
+
+	public Geometry exportTriangles(GeometryFactory gf) {
+		if (sd != null) {
+			return sd.getTriangles(gf);
+		} else {
+			return null;
+		}
+	}
+
+	/** dig depth-first, faster but may generate weird results
+	 * @param triChecker Triangle Checker to be used
+	 * @return concave hull of input data
+	 */
+	public Geometry getConcaveHullDFS(TriangleChecker triChecker) {
+		if (triChecker != null) {
+			TreeSet<Coordinate> nodeSet = new TreeSet<Coordinate>(hullDT);
+			LinkedList<Coordinate> hullList = new LinkedList<Coordinate>(hullDT);
+			boolean modified = true;
+			while (modified) {
+				modified = false;
+				ListIterator<Coordinate> iter = hullList.listIterator(0);
+				while (iter.hasNext()) {
+					Coordinate nodeS = iter.next(); // iter after nodeS
+					if (iter.hasNext()) {
+						Coordinate nodeE = iter.next(); // iter afer nodeE
+						QuadEdge qe = sd.locate(nodeS, nodeE);
+						if (qe == null) {
+							System.out.println("fail to find edge: " + nodeS.toString() + " -" + nodeE.toString());
+							return null;
+						}
+						QuadEdge lnext = qe.lNext();
+						Vertex verO = lnext.dest();
+						Coordinate nodeO = verO.getCoordinate();
+						if (nodeSet.contains(nodeO)) {
+							iter.previous();
+							continue; // will break areal connectivity so omit?
+						}
+						if (triChecker.removeable(nodeS, nodeE, nodeO)) {
+							modified = true;
+							nodeSet.add(nodeO);
+							iter.previous(); // move iter before nodeE
+							iter.add(nodeO); // insert nodeO between nodeS and nodeE (and before iter)
+							iter.previous(); // iter before nodeO
+							iter.previous(); // iter before nodeS
+						} else {
+							iter.previous(); // iter before nodeE
+						}
+					}
+				}
+			}
+			//
+			Coordinate[] newCoords = new Coordinate[hullList.size()];
+			hullList.toArray(newCoords);
+			return gf.createPolygon(newCoords);
+		} else {
+			return null;
+		}
+	}
+
+	/** breadth first digging, supports multi parts
+	 * @param triChecker triangle checker to be used
+	 * @param allowMultiParts if multiple parts are to be generated
+	 * @param keepLineSeg if degenerated line segments should be kept
+	 * @return a collection of geometry that form the concave hull of the input data (may contains linestring as degenerated segments)
+	 */
+	public Collection<Geometry> getConcaveHullBFS(TriangleChecker triChecker, boolean allowMultiParts, boolean keepLineSeg) {

```

Comment:
PLease refactor this method into several shorter ones.

Reply:
Removed the whole package as it is not used. It supported holes for isochrones, but this was not tested and adjusted thoroughly enough imo. Also unnecessary at the moment. Can be re-added if ever wanted.